### ISSUE TITLE
Review RH NSM operator

### ISSUE THREAD

**Issue by johlj** [Nordix/Meridio#136]
https://github.com/networkservicemesh/nsm-operator/pull/7

**Comment by mandydydy** [comment]
Some comments:

- Must Spire helm chart be stored in the operator?

- In NSM CRD, the full image paths are split into registry, organization, and image names, of which the first two are shared. This either provides insufficient flexibility for users to save images in their own registries or provides too many parameters if users are only allowed to use “official” images from the NSM registry. 

- Why apiextensions version in CRD is downgraded from v1 to v1beta1? V1beta1 is no longer served as of v1.22 https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122

**Comment by acmenezes** [comment]
> * Must Spire helm chart be stored in the operator?

No. The Spire helm chart doesn't need to be stored in the operator. That chart was provided to me by Ericsson engineers I must know if the chart is exactly the same provided by Spire itself. We can totally use whatever official chart we have. I'm going to check the official one and make new tests with that. We can totally discard that one if the tests succeed.

**Comment by acmenezes** [comment]
> * In NSM CRD, the full image paths are split into registry, organization, and image names, of which the first two are shared. This either provides insufficient flexibility for users to save images in their own registries or provides too many parameters if users are only allowed to use “official” images from the NSM registry.

@mandydydy I got your point. Seems right. Do you have any suggestions on how we should deal with it? What would be the best combination in that case?

**Comment by acmenezes** [comment]
> * Why apiextensions version in CRD is downgraded from v1 to v1beta1? V1beta1 is no longer served as of v1.22 https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122

In fact that has been already noted before. I will add the upgrade to that PR.

**Comment by mandydydy** [comment]
> > * In NSM CRD, the full image paths are split into registry, organization, and image names, of which the first two are shared. This either provides insufficient flexibility for users to save images in their own registries or provides too many parameters if users are only allowed to use “official” images from the NSM registry.
> 
> @mandydydy I got your point. Seems right. Do you have any suggestions on how we should deal with it? What would be the best combination in that case?

How about using the version as the minimum required parameter, with image paths(registry+organization but without version) being optional parameters? If only the version is specified, use the official image paths as default, otherwise use the specified image paths.

**Comment by acmenezes** [comment]
@mandydydy finally done. So:
1) the official spire helm chart doesn't actually exist. So for now I think we can merge it like that to ease up the install procedure until we can create a better and more elegant solution. Tell what you think. I'm open to suggestions as well.
2) api extensions a.k.a. CRDs are now in version V1
3) The fields registry and organization are now optional and are filled with the official network service mesh values if they come empty. Only the version field is required now. The tag field was replaced by the version field. Check the new Spec and the new sample CR as well.
OBS: I also tested with NSM v1.1.0.  Suggested by Ed Warnick.

Please let me know what you think. If it's cool give a lgtm if possible on the service mesh repo. Thanks a lot for your review. It's really appreciated.
